src/Microsoft.AspNetCore.Authentication.JwtBearer/JwtBearerHandler.cs

```diff
@@ -191,6 +191,13 @@ internal class JwtBearerHandler : AuthenticationHandler<JwtBearerOptions>
             {
                 AuthenticateFailure = authResult?.Failure,
             };
+
+            if (Options.IncludeErrorDetails && (authResult != null && !authResult.Skipped))
+            {
+                eventContext.Error = "invalid_token";
```

Comment:
This sets `error` even when there was no Failure. Auth may have been successful and the app could choose to return 401 anyways. Check for Failure rather than Skipped.